x/trigger/keeper/event_listener.go

```diff
@@ -1,58 +1,108 @@
 package keeper
 
 import (
-	"encoding/binary"
+	"crypto/sha256"
+	"errors"
+	"strings"
 
-	storetypes "cosmossdk.io/store/types"
+	"cosmossdk.io/collections"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	triggertypes "github.com/provenance-io/provenance/x/trigger/types"
 )
 
 // SetEventListener Adds the trigger to the event listener store.
-func (k Keeper) SetEventListener(ctx sdk.Context, trigger triggertypes.Trigger) {
-	store := ctx.KVStore(k.storeKey)
-	event, _ := trigger.GetTriggerEventI()
-	store.Set(triggertypes.GetEventListenerKey(event.GetEventPrefix(), event.GetEventOrder(), trigger.GetId()), []byte{})
+func (k Keeper) SetEventListener(ctx sdk.Context, trigger triggertypes.Trigger) error {
+	event, err := trigger.GetTriggerEventI()
+	if err != nil {
+		return err
+	}
+	eventHash, err := getEventNameHash(event.GetEventPrefix())
+	if err != nil {
+		return err
+	}
+	key := collections.Join3(eventHash, event.GetEventOrder(), trigger.GetId())
```

Comment:
This is done the same way in a lot of places. Its probably best to just refactor `GetEventListenerKey` to return the `collections.Join3` result.